core/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -131,6 +142,71 @@ public Task unregister(Task task) {
     }
 
     /**
+     * Stores the task failure
+     */
+    public <Response extends  ActionResponse> void persistResult(Task task, Throwable error, ActionListener<Response> listener) {
+        DiscoveryNode localNode = lastDiscoveryNodes.getLocalNode();
+        if (localNode == null) {
+            // too early to persist anything, shouldn't really be here - just pass the error along
+            listener.onFailure(error);
+            return;
+        }
+        final TaskResult taskResult;
+        try {
+            taskResult = task.result(localNode, error);
+        } catch (IOException ex) {
+            logger.warn("couldn't persist error ", error);
```

Comment:
Maybe you should includes `ex` in the WARN here and still pass the original error along? Or stick the original error in `addSuppressedException`?